docs/design/fabric-v2+/fpc-registration.puml

```diff
@@ -12,261 +12,155 @@
 
 !pragma teoz true
 
-hide footbox
-title Foot Box removed
+'hide footbox
+'title Foot Box removed
 
-title Fabric Private Chaincode - Enclave registry - New approach
+title FPC Enclave Registration \n ***** DOCUMENT IN PROGRESS ******
 
-actor "User Org1" as User1 order 9
 actor "Admin Org1" as Admin order 10
-participant "ECC_Manager (syscc) Peer1_Org1" as Manager1 order 20
+participant "ECC Peer1_Org1" as ECC1 order 20
+participant "ECC_Enclave" as Enclave1 order 30 #99FF99
+participant "ERCC Peer1_Org1" as ERCC1 order 40
+participant "ERCC Peer1_Org2" as ERCC2 order 50
 
-database "Local storage Peer1_Org1" as storage1 order 24
-participant "TLCC_Enclave Peer1_Org1" as TLCC1 order 25 #99FF99
+entity IAS order 1
 
-participant "ECC Peer1_Org1" as ECC1 order 30
-
-participant "ERCC Peer1_Org1" as ERCC1 order 50
-
-participant "ECC Peer1_Org2" as ECC2 order 60
-participant "ERCC Peer1_Org2" as ERCC2 order 62
-
-participant "ECC Peer1_Org3" as ECC3 order 70
-participant "ERCC Peer1_Org3" as ERCC3 order 72
-
-entity Orderer order 100
-
-note left ERCC1
-  ERCC can be **integrated** in ECC or as **seperate** chaincode
-  Integrated:
-  - ERCC and ECC share endorsement policy (but need different "type": org-based vs. attestation-based)
-  - attestations stored ECC namespace
-  Separate: <- we use this one
-  - flexible endorsement policies (ERCC and ECC can have different)
-  - attestations stored in ERCC namespace
-end note
-
-note left ERCC1
-  FPC chaincode has three different associated endorsement policies (EP):
-  - lifecycle transactions ("normal fabric": approve chaincode definition (define application transaction EP), upgrade, etc ...)
-  - application transactions
-  - FPC-lifecycle transactions (register, key-gen, key-dist, ...)
-
-  Decision: We use a **separate** ERCC.
-  We realized FPC-liflecycle as application transaction EP of ERCC chaincode;
-  should ideally match lifecycle EP which is 'Channel/Application/LifecycleEndorsement' and by default majority of organizations)
```

Comment:
as discussed in phoenix, we should discuss somewhere the involved endorsement policies and the requirement of consistency between the EP errc uses and the Lifecyle EP used by fpc chaincode creation.  Insofar some of these comments would still be useful?

Reply:
added a comment.